### PR TITLE
[RW-11461] Add withSpanResource helper function and use in AppRoutes and AppRoutesV2

### PR DESCRIPTION
**Pull request by jmthibault79**
__Jira ticket__: https://precisionmedicineinitiative.atlassian.net/browse/RW-11461

### What

No expected changes in functionality.  This is a code refactoring to simplify adding `spanResource` to API calls.

I updated AppRoutes and AppRoutesV2 to use this new function, but the change can be applied to all routes.  Should I do so here?

### Why

-

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

**Review by Qi77Qi — APPROVED** [review]

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/AppRoutes.scala [hunk 1]
```diff
@@ -141,92 +132,88 @@ class AppRoutes(kubernetesService: AppService[IO], userInfoDirectives: UserInfoD
...
+  ): IO[ToResponseMarshallable] = {
+    val apiCallName = "getApp"
     for {
-      ctx <- ev.ask[AppContext]
-      apiCall = kubernetesService.getApp(
-        userInfo,
-        CloudContext.Gcp(googleProject),
-        appName
+      _ <- metrics.incrementCounter(apiCallName)
+      resp <- withSpanResource(apiCallName,
```
Comment (Qi77Qi) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/AppRoutes.scala:151]: can potentially put the metrics line into the `withSpanResource` too...

**Review by Qi77Qi — APPROVED** [review]

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/AppRoutes.scala (hunk 1 above)
Comment (jmthibault79) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/AppRoutes.scala:151]: I was thinking of that. However, not all routes have a metrics call like this.  Sometimes metrics are called from the service instead.

**Review by mlilynolting — APPROVED** [review]
Approved, but I like Qi's idea of folding the metrics call into `withSpanResource`

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/AppRoutes.scala (hunk 1 above)
Comment (jmthibault79) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/AppRoutes.scala:151]: see also `createAppHandler()` where the metrics have some associated logic

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/AppRoutes.scala:151]: ah right, I think for some, we needed additional info, so we had to move it to service layer...

**Comment by jmthibault79** [comment]
Azure tests failed: https://github.com/broadinstitute/terra-github-workflows/actions/runs/9370192130/job/25796565555
```
[info] *** 1 TEST FAILED ***
Error:  Failed tests:
Error:  	org.broadinstitute.dsde.workbench.leonardo.LeonardoAzureSuite
```

Rerunning.

**Pull request merged**